tasks/Task.cpp

```diff
@@ -50,27 +51,38 @@ bool Task::configureHook()
     if (!m_handle) {
         throw runtime_error("Failed to connect to sensor!");
     }
+    m_scan_timeout = _scan_timeout.get();
+    m_first_scan_timeout = _first_scan_timeout.get();
     m_packet_buffer.resize(m_udp_buf_size);
     m_vertical_fov = _vertical_fov.get();
     m_remission_enabled = _remission_enabled.get();
-    return true;
-}
-bool Task::startHook()
-{
-    if (!TaskBase::startHook())
-        return false;
     m_metadata = getMetadata();
     configureDepthMap();
     m_packet_format.reset(new packet_format(get_format(m_metadata)));
     // A ScanBatcher can be used to batch packets into scans
     m_scan_batcher.reset(
         new ScanBatcher(m_metadata.format.columns_per_frame, *m_packet_format));
+    auto data = acquireData(m_first_scan_timeout);
+    if (!data.complete(m_metadata.format.column_window)) {
+        exception(TIMEOUT);
+        return false;
+    }
+    return true;
+}
+bool Task::startHook()
+{
+    if (!TaskBase::startHook())
+        return false;
     return true;
 }
 void Task::updateHook()
 {
     TaskBase::updateHook();
-    auto data = acquireData();
+    auto data = acquireData(m_scan_timeout);
+    if (!data.complete(m_metadata.format.column_window)) {
+        exception(TIMEOUT);
```

Comment:
Do you think one exception that says TIMEOUT is enough, or two, one for the first scan, and the other for it on the updataHook?

Reply:
It will be clear if it happens in configure or update hook, so I think this modification isn't needed